src/mlpack/methods/adaboost/adaboost_main.cpp

```diff
@@ -205,6 +206,10 @@ static void mlpackMain()
       trainingData.shed_row(trainingData.n_rows - 1);
     }
 
+    // Sanity check on data and labels
+    util::CheckSameSizes(trainingData, labelsIn, 
```

Comment:
What I had in mind over here was to create a check similar to
[this](https://github.com/mlpack/mlpack/blob/70f0a4b82c26ab841cfba2d90835fe91204471a9/src/mlpack/core/util/param_checks_impl.hpp#L2) . To define that check you can use the 
CheckSameSizes and CheckSameDimensionality utilities. 
This seems like a better way to do this, as it will keep uniformity in the code.